zebrad/src/commands/start/sync.rs

```diff
@@ -174,22 +170,20 @@ where
                         // It indicates that the remote peer does not have any blocks
                         // following the prospective tip.
                         // TODO(jlusby): reject both main and test net genesis blocks
-                        if hashes[0] == super::GENESIS {
-                            tracing::debug!("skipping response that does not extend the tip");
-                            continue;
+                        match hashes.first() {
+                            Some(&super::GENESIS) => {
+                                tracing::debug!("skipping response that does not extend the tip");
+                                continue;
+                            }
+                            Some(_) | None => {}
```

Comment:
I think this has the same problem as here: https://github.com/ZcashFoundation/zebra/pull/531#discussion_r445117781

Reply:
This will be fixed when https://github.com/ZcashFoundation/zebra/pull/543 merges and I rebase this

Reply:
I think it would be better to just keep the previous logic.